portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -74,14 +76,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = request.getParameter("name");
+    // only logged-in users with nicknames can comment
+    UserService userService = UserServiceFactory.getUserService();
+
+    String email = userService.getCurrentUser().getEmail();
```

Comment:
Create a `currentUser` variable for `userService.getCurrentUser()` since it’s been called multiple times.